qldpc/codes_test.py

```diff
@@ -114,37 +114,62 @@ def test_CSS_shifts(
     """Apply Pauli deformations to a CSS code."""
     code_a = codes.ClassicalCode.random(*bits_checks_a)
     code_b = codes.ClassicalCode.random(*bits_checks_b)
-    matrix_x, matrix_z, _ = codes.HGPCode.get_hyper_product(code_a, code_b)
-
+    matrix_x, matrix_z = codes.HGPCode.get_hyper_product(code_a, code_b)
     num_qubits = matrix_x.shape[-1]
+    code = codes.CSSCode(matrix_x, matrix_z)
     conjugate = tuple(qubit for qubit in range(num_qubits) if np.random.randint(2))
     shifts = {qubit: np.random.randint(3) for qubit in range(num_qubits)}
-    code = codes.CSSCode(
-        matrix_x,
-        matrix_z,
-        qubits_to_conjugate=conjugate,
-        qubit_shifts=shifts,
-    )
+    print(conjugate)
+    transformed_matrix = codes.CSSCode.conjugate(code.matrix, conjugate)
+    transformed_matrix = codes.CSSCode.shift(transformed_matrix, shifts)
 
     for node_check, node_qubit, data in code.graph.edges(data=True):
         pauli_index = np.where(data[codes.Pauli].value)
-        assert (code.matrix[node_check.index, pauli_index, node_qubit.index] == 1).all()
+        assert (transformed_matrix[node_check.index, pauli_index, node_qubit.index] == 1).all()
+
+
+def test_Qudit_stabs(
+    bits_checks_a: tuple[int, int] = (5, 3),
+    bits_checks_b: tuple[int, int] = (3, 2),
+) -> None:
+    code_a = codes.ClassicalCode.random(*bits_checks_a)
+    code_b = codes.ClassicalCode.random(*bits_checks_b)
+    matrix_x, matrix_z, _ = codes.HGPCode.get_hyper_product(code_a, code_b)
+    code = codes.CSSCode(matrix_x, matrix_z)
+    stabs = code.matrix_to_stabilizers()
+    assert len(stabs) == code.num_checks
+    # for stab in stabs :
+    #    print(stab)
+    # print(code.matrix)
+    return
+
+
+def test_Qudit_graph(
+    bits_checks_a: tuple[int, int] = (5, 3),
+    bits_checks_b: tuple[int, int] = (3, 2),
+) -> None:
+    code_a = codes.ClassicalCode.random(*bits_checks_a)
+    code_b = codes.ClassicalCode.random(*bits_checks_b)
+    matrix_x, matrix_z, _ = codes.HGPCode.get_hyper_product(code_a, code_b)
+    code = codes.CSSCode(matrix_x, matrix_z)
+    assert np.array_equal(code.graph_to_matrix(code.graph), code.matrix)
+    return
 
 
 @pytest.mark.parametrize("conjugate", [False, True])
 def test_trivial_lift(
-    conjugate: bool,
     bits_checks_a: tuple[int, int] = (10, 8),
     bits_checks_b: tuple[int, int] = (7, 3),
+    field: int = 2,
```

Comment:
May as well test with something more interesting
```suggestion
    field: int = 3,
```